lib/publiccloud/provider.pm

```diff
@@ -41,10 +41,34 @@ sub init {
     my ($self) = @_;
     my $file = lc get_var('PUBLIC_CLOUD_PROVIDER');
     assert_script_run('mkdir -p ' . TERRAFORM_DIR);
-    assert_script_run('curl ' . data_url('publiccloud/terraform/' . $file . '.tf') . ' -o ' . TERRAFORM_DIR . '/plan.tf');
+    if (check_var('SLES4SAP_MODE', 'publiccloud')) {
+        my $git_repo   = get_var('ha-sap-terraform-deployments');
+        my $cloud_name = $self->get_sap_cloud_name;
+        assert_script_run('cd ' . TERRAFORM_DIR);
+        assert_script_run('git clone --depth 1 --branch ' . get_var('HA_SAP_GIT_TAG', 'master') . ' ' . get_required_var('HA_SAP_GIT_REPO') . ' .');
+        assert_script_run('cp pillar_examples/automatic/{cluster.sls,hana.sls,top.sls} salt/hana_node/files/pillar/');
+        assert_script_run('cd');    # We need to ensure to be in the home directory
+        assert_script_run('curl ' . data_url("publiccloud/terraform/sap/$file.tfvars") . ' -o ' . TERRAFORM_DIR . "/$cloud_name/terraform.tfvars");
+        $self->create_ssh_key(ssh_private_key_file => TERRAFORM_DIR . '/salt/hana_node/files/sshkeys/cluster.id_rsa');
+    }
+    else {
+        assert_script_run('curl ' . data_url("publiccloud/terraform/$file.tf") . ' -o ' . TERRAFORM_DIR . '/plan.tf');
+    }
     $self->create_ssh_key();
```

Comment:
in case of SLES4SAP_MODE you calling ssh key creation two times ?

Reply:
Yes, as we need SSH keys dedicated for the HA part. And a good practice is to not use the same keys as root.